spec/models/package_spec.rb

```diff
@@ -491,6 +491,75 @@
       expect(package.reload.received_quantity).to eq received_quantity
     end
   end
+
+  describe '#create_associated_packages_location' do
+    let(:package) { create :package }
+    let(:location) { create :location }
+
+    it 'creates associated package location record for package' do
+      expect{
+        package.create_associated_packages_location(location.id, 2)
+      }.to change(PackagesLocation, :count).by(1)
+    end
+  end
+
+  describe '#create_or_update_location_for_dispatch_from_stockit' do
+    let(:package) { create :package }
+    let(:order) { create :order }
+    let(:orders_package) { create :orders_package, state: 'dispatched', package: package, order: order }
+    let(:dispatched_location) { create :location, :dispatched }
+
+    it 'updates orders_package_id against packages_location record if dispatched' do
+      packages_location = create :packages_location, package: package, location: dispatched_location
+      package.create_or_update_location_for_dispatch_from_stockit(dispatched_location, orders_package.id, orders_package.quantity)
+      expect(packages_location.reload.reference_to_orders_package).to eq orders_package.id
+    end
+
+    it 'creates new packages_location record with orders_package_id if packages_location record do not exist and package dispatched' do
+      expect{
+        package.create_or_update_location_for_dispatch_from_stockit(dispatched_location, orders_package.id, orders_package.quantity)
+      }.to change(PackagesLocation, :count).by(1)
+      expect(package.reload.packages_locations.first.reference_to_orders_package).to eq orders_package.id
+    end
+  end
+
+  describe '#create_dispatched_packages_location_from_gc' do
+    let(:dispatched_location) { create :location, :dispatched }
+    let(:order) { create :order }
+    let(:orders_package) { create :orders_package, state: 'dispatched', package: package, order: order }
+    let(:dispatched_location) { create :location, :dispatched }
+
+    it 'creates dispatched packages location record against package if do not exist' do
+      expect{
+        package.create_dispatched_packages_location_from_gc(dispatched_location, orders_package.id, 1)
+      }.to change(PackagesLocation, :count).by(1)
+      expect(package.reload.packages_locations.first.location).to eq dispatched_location
```

Comment:
perhaps remove repetition here:

    first_location = package.reload.package_locations.first.location
    expect(first_location).to eq(...)